packages/edit-site/src/components/global-styles/screen-revisions/revisions-buttons.js

```diff
@@ -6,28 +6,91 @@ import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { __, sprintf } from '@wordpress/i18n';
+import { __, _n, sprintf } from '@wordpress/i18n';
 import { Button } from '@wordpress/components';
 import { dateI18n, getDate, humanTimeDiff, getSettings } from '@wordpress/date';
 import { store as coreStore } from '@wordpress/core-data';
 import { useSelect } from '@wordpress/data';
+import { useMemo } from '@wordpress/element';
+import { getBlockTypes } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import getRevisionChanges from './get-revision-changes';
 
 const DAY_IN_MILLISECONDS = 60 * 60 * 1000 * 24;
+const MAX_CHANGES = 7;
+
+function ChangedSummary( { revision, previousRevision, blockNames } ) {
+	let changes = getRevisionChanges( revision, previousRevision, blockNames );
+
+	const changesLength = changes.length;
+
+	if ( ! changesLength ) {
+		return null;
+	}
+
+	// Truncate to `n` results.
+	if ( changesLength > MAX_CHANGES ) {
+		changes = changes.slice( 0, MAX_CHANGES );
+	}
+
+	const moreChanges = changesLength - changes.length;
+
+	changes = changes.reduce( ( acc, curr ) => {
+		acc[ curr[ 0 ] ] = ! acc[ curr[ 0 ] ]
+			? [ curr[ 1 ] ]
+			: [ ...acc[ curr[ 0 ] ], curr[ 1 ] ];
+		return acc;
+	}, {} );
+
+	return (
+		<div className="edit-site-global-styles-screen-revision__changes">
```

Comment:
Note to self: probably can do with a little clean up, maybe rethink HTML elements.